comp110-worksheet-6/OxoBoard.cs

```diff
@@ -12,36 +12,75 @@ public class OxoBoard
 	{
 		// Constructor. Perform any necessary data initialisation here.
 		// Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
-		public OxoBoard(/* int width = 3, int height = 3, int inARow = 3 */)
+		public OxoBoard(/*int width = 3, int height = 3, int inARow = 3 */)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            playerBoard = new Mark[3, 3];
 		}
 
 		// Return the contents of the specified square.
-		public Mark GetSquare(int x, int y)
+		public Mark GetSquare(int a, int b)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+			return playerBoard[a, b];
 		}
 
 		// If the specified square is currently empty, fill it with mark and return true.
 		// If the square is not empty, leave it as-is and return False.
-		public bool SetSquare(int x, int y, Mark mark)
+		public bool SetSquare(int a, int b, Mark mark)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            if (playerBoard[a, b] == Mark.None)
+            {
+                playerBoard[a, b] == mark;
+                return true;
+            }
+            else
+            {
+                return false;
+            
+            }
 		}
 
 		// If there are still empty squares on the board, return false.
 		// If there are no empty squares, return true.
 		public bool IsBoardFull()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            for (int a = 0; a <  playerBoard.GetLength(0); a++)
+            {
+                for (int b = 0; b < playerBoard.GetLength(0); b++)
```

Comment:
This should be `GetLength(1)` (both happen to be the same number in this case but it's best to be consistent)